web/styles/zulip.css

```diff
@@ -1049,6 +1049,110 @@ p.n-margin {
     position: relative;
 }
 
+/* Styles for the overlay screen that appears while uploading files using drag-and-drop */
+.upload-overlay-container {
+    visibility: hidden;
+    position: fixed;
+    display: flex;
+    top: 0;
+    left: 0;
+    width: 100vw;
+    height: 100vh;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    gap: 16px;
+    /* Semi-transparent background with a blur effect */
+    background: hsl(238deg 35% 15% / 65%);
+    backdrop-filter: blur(8px);
+    transition: all 400ms;
+    opacity: 0;
+    /* Initial low z-index to avoid interfering with other components */
+    z-index: -1;
+}
+
+/* Dynamic class to make the overlay visible during drag events */
+.upload-overlay-visibility {
+    visibility: visible;
+    /* Higher z-index to overlay other components during file upload */
+    z-index: 10000;
+    opacity: 1;
+    pointer-events: all;
+}
+
+.upload-overlay-svg-container {
+    width: 100%;
+    height: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    margin: auto;
+}
+
+/* Styles for the animated border SVG */
+.upload-overlay-border {
+    width: calc(100% - 80px);
+    height: calc(100% - 80px);
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    @media (width <= 400px) {
+        width: calc(100% - 20px);
+        height: calc(100% - 20px);
+    }
+}
+
+.upload-overlay-border rect {
+    width: calc(100% - 3px);
```

Comment:
Comments here, too. It's a good idea to help along future contributors by explaining where these kinds of exacting values are coming from.

Reply:
Same, because of the issue description and figma design.